File: server/Models/UserAddress.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Backend.Models
{
    public class UserAddress
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public int UserId { get; set; }
        [Required]
        [StringLength(200)]
        public string Address { get; set; } = null!;
        [Required]
        [StringLength(100)]
        public string City { get; set; } = null!;
        [Required]
        [StringLength(100)]
        public string District { get; set; } = null!;
        [Required]
        [StringLength(100)]
        public string Ward { get; set; } = null!;
        [Required]
        [StringLength(100)]
        public string Country { get; set; } = null!;
     
        public string? PostalCode { get; set; }

        public User User { get; set; } = null!;

    }
}

File: mobile/src/types/userAddress.ts
import { BaseModel } from "../base/model";

export interface UserAddress extends BaseModel {
  userId: number;
  address: string | null;
  city: string | null;
  district: string | null;
  ward: string | null;
  country: string | null;
  postalCode?: string | null;
}

export type UserAddressRequest = {
  userId: number;
  address: string;
  city: string;
  district: string;
  ward: string;
  country: string;
  postalCode?: string;
};
